extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -188,6 +202,7 @@ export class AmpAd3PImpl extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    this.lifecycleReporter_.sendPing('layoutCallback');
```

Comment:
I totally understand why we want to report those numbers. My suggestion is about the event names and possibly better code point to report for some of the events. Instead of using the function names, event names like adRequestStart, renderStart would make a lot more sense here. layoutCallback, onLayoutMeasure, are really implementation details, we don't want the graph to be changed when we simply moved some logic from one method to another. Good names also help us to do fair comparison between 3p and a4a.